pkg/operator/controller/ingress/status_test.go

```diff
@@ -959,6 +959,46 @@ func TestComputeIngressAvailableCondition(t *testing.T) {
 	}
 }
 
+func TestComputeDeploymentProgressingCondition(t *testing.T) {
+	deployment := func(conditionType appsv1.DeploymentConditionType, conditionStatus corev1.ConditionStatus) *appsv1.Deployment {
+		return &appsv1.Deployment{
+			Status: appsv1.DeploymentStatus{
+				Conditions: []appsv1.DeploymentCondition{{
+					Type:   conditionType,
+					Status: conditionStatus,
+				}},
+		}}
+	}
+	testCases := []struct {
+		description string
+		deployment  *appsv1.Deployment
+		expect      operatorv1.OperatorCondition
+	}{
+		{
+			description: "deployment has a progressing condition",
+			deployment: deployment(appsv1.DeploymentProgressing, corev1.ConditionTrue),
+			expect: operatorv1.OperatorCondition{Type: IngressControllerDeploymentProgressingConditionType, Status: operatorv1.ConditionTrue},
```

Comment:
Should we also assert on the contents of `Reason` and `Message`?

Reply:
I think that'd be a good idea. I noticed that Deployments have a nuance where they show progressing even after the replicaset has finished progressing so testing that the reason and message are expected makes sense.

Reply:
Same here; rewritten in the latest push